src/config_parser.rs

```diff
@@ -0,0 +1,119 @@
+use config::{Config, ConfigError, ValueKind};
+
+use crate::pipewire_listener::{PortConnections, StreamConnections};
+
+#[allow(dead_code)]
+#[derive(Debug, Clone)]
+pub struct TurboAudioConfig {
+    pub device_name: Option<String>,
+    pub jack: bool,
+    pub sample_rate: i64,
+    pub stream_connections: Vec<StreamConnections>,
+}
+
+impl TurboAudioConfig {
+    fn new() -> Self {
+        Self {
+            device_name: None,
+            jack: false,
+            sample_rate: 48000,
+            stream_connections: vec![],
+        }
+    }
+}
+
+pub fn parse_config(config_name: &str) -> anyhow::Result<TurboAudioConfig> {
+    let mut config = TurboAudioConfig::new();
+
+    let settings = Config::builder()
+        .add_source(config::File::with_name(config_name))
+        .build()
+        .unwrap();
```

Comment:
how can this fail?

Reply:
Ah ouais cest si ya pas de fichier de config. Je devrais handle nice catch

Reply:
Ok j'ai mis que si tas pas de config file ca fait juste default a tout et ca le log LMK si cest good